src/main/java/frontend/src/ChatPage.tsx

```diff
@@ -1,50 +1,64 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 import './ChatPage.css';
 
 interface Message {
-    id: number;
-    text: string;
+    id: string;
     sender: string;
+    receiver: string;
+    content: string;
+    timestamp: number;
 }
 
 const ChatPage: React.FC = () => {
     const [messages, setMessages] = useState<Message[]>([]);
     const [newMessage, setNewMessage] = useState('');
+    const [receiver, setReceiver] = useState('');
+    const sender = 'currentUser'; // This should be replaced with the actual current user
 
-    const handleSendMessage = () => {
-        if (newMessage.trim() === '') return;
-
-        const message: Message = {
-            id: messages.length + 1,
-            text: newMessage,
-            sender: 'User',
+    useEffect(() => {
+        const fetchMessages = async () => {
+            const response = await axios.get(`${process.env.REACT_APP_API_URL}/messages/${sender}/${receiver}`);
+            setMessages(response.data);
         };
 
+        if (receiver) {
+            fetchMessages();
+        }
+    }, [receiver]);
+
+    const handleSendMessage = async () => {
+        const message = { sender, receiver, content: newMessage, timestamp: Date.now() };
```

Comment:
Wouldn't it be better to set the timestamp on the server? Your client could post ANY time (just set a different time on your machine)